apps/dapp/src/providers/SwapProvider.tsx

```diff
@@ -1,125 +1,98 @@
-import { useState, useContext, createContext, PropsWithChildren } from 'react';
-import { BigNumber, Signer, utils } from 'ethers';
+import { useState, useContext, createContext, PropsWithChildren, useEffect } from 'react';
+import { BigNumber, Contract, ethers } from 'ethers';
 import { TransactionReceipt } from '@ethersproject/abstract-provider';
 import { useWallet } from 'providers/WalletProvider';
 import { useNotification } from 'providers/NotificationProvider';
 import { SwapService } from 'providers/types';
-import { NoWalletAddressError } from 'providers/errors';
 import { TICKER_SYMBOL } from 'enums/ticker-symbol';
 import { asyncNoop } from 'utils/helpers';
-import { fromAtto } from 'utils/bigNumber';
-import { ERC20__factory, TempleERC20Token__factory, TempleUniswapV2Pair__factory } from 'types/typechain';
+import { ERC20__factory, TempleERC20Token__factory } from 'types/typechain';
 import env from 'constants/env';
 import { AnalyticsEvent } from 'constants/events';
 import { AnalyticsService } from 'services/AnalyticsService';
 import { formatBigNumber, getTokenInfo } from 'components/Vault/utils';
+import { SwapInfo } from '@balancer-labs/sor';
+import { BalancerSDK, Network, SwapType } from '@balancer-labs/sdk';
+import vaultArtifact from 'data/abis/balancerVault.json';
+import { formatToken } from 'utils/formatter';
+
+// Initialize balancer SOR
+const gasPrice = BigNumber.from('14000000000');
+const maxPools = 4;
+const balancer = new BalancerSDK({
```

Comment:
I haven't found a good way to test on goerli yet, will add this as an open issue